test/unit/test_exports/test_derived_quantities/test_maximum_surface.py

```diff
@@ -38,17 +40,23 @@ class TestCompute:
     my_max.function = c
     my_max.dx = dx
 
-    def test_minimum(self):
+    def test_maximum(self):
         dm = self.V.dofmap()
-        subd_dofs = np.unique(
-            np.hstack(
-                [
-                    dm.cell_dofs(c.index())
-                    for c in f.SubsetIterator(self.surface_markers, self.surface)
-                ]
-            )
+        facets = self.surface_markers.where_equal(self.surface)
+        entity_closure_dofs = np.array(
+            dm.entity_closure_dofs(self.mesh, self.mesh.topology().dim() - 1, facets),
+            dtype=np.int32,
         )
-        expected = np.max(self.c.vector().get_local()[subd_dofs])
+        local_dofs = entity_closure_dofs < len(self.c.vector().get_local())
+        if len(local_dofs) == 0:
+            local_max = -np.inf
+        else:
+            local_max = np.max(
+                self.c.vector().get_local()[entity_closure_dofs[local_dofs]]
+            )
+
+        expected = MPI.max(self.mesh.mpi_comm(), local_max)
```

Comment:
Since here, the function is simply $y=x$, we know what the expected maximum is without having to compute it. It would make this test more straightforward